### PR TITLE
interpreter: (riscv) Add some libc runtime methods like malloc/free

### PR DESCRIPTION
**Pull request by AntonLydike**
Uses the interpreter external functions from #1536 to implement malloc, calloc, free and putchar.

### CODE REVIEW

File: tests/filecheck/runner/riscv.mlir [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// RUN: xdsl-run %s
```
Comment (superlopuh) [inline tests/filecheck/runner/riscv.mlir:1]: can you add the filecheck bit, please?

**Review by superlopuh — CHANGES_REQUESTED** [review]
beautiful, just one minor change remaining

File: tests/filecheck/runner/riscv.mlir (hunk 1 above)
Comment (AntonLydike) [reply tests/filecheck/runner/riscv.mlir:1]: That would indeed be quite useful to have

Comment (AntonLydike) [reply tests/filecheck/runner/riscv.mlir]: Fixed!

**Review by superlopuh — APPROVED** [review]
thank you

**Pull request merged**